core/modules/installer/installer.pages.inc

```diff
@@ -921,6 +921,7 @@ function installer_browser_filters_form_submit($form, &$form_state) {
   }
   else {
     $_SESSION['installer_browser_text_filter_' . $type] = $form_state['values']['search_text'];
+    unset($_GET['page']);
```

Comment:
From the issue:

> The "Clear" button doesn't reset the page... which isn't broken as it just shows different modules from whatever page you happen to be on, but is an odd bit of UX.

So seems like you'd want to clear the page query string regardless of whether the clear or filter buttons were pressed. It should always be reset to the first page if the user performs any filtering action at all.